blueman/gui/manager/ManagerDeviceList.py

```diff
@@ -221,16 +218,16 @@ def device_remove_event(self, device):
         tree_iter = self.find_device(device)
 
         row_fader = self.get(tree_iter, "row_fader")["row_fader"]
-        self._faderhandler = row_fader.connect("animation-finished", self.__on_fader_finished, device, tree_iter)
+        super().device_remove_event(device)
+        row_fader.connect("animation-finished", self.__on_fader_finished)
+
         row_fader.thaw()
         self.emit("device-selected", None, None)
         row_fader.animate(start=row_fader.get_state(), end=0.0, duration=400)
 
-    def __on_fader_finished(self, fader, device, tree_iter):
-        fader.disconnect(self._faderhandler)
-        self._faderhandler = None
+    def __on_fader_finished(self, fader):
+        fader.disconnect_by_func(self.__on_fader_finished)
```

Comment:
disconnect_by_func is broken (kindof see https://gitlab.gnome.org/GNOME/pygobject/-/issues/106) so don't use it.

I don't have time to look at the PR right now maybe in the weekend I can actually have a proper look.

Reply:
I know about that bug, but it does not effect us as we only ever attach one fader anyway. I did check if the handler is correctly disconnected by disconnect_by_func: it is.

Reply:
Today it may cause no problems but tomorrow we change things and wonder why  it is broken. I have been bitten by this before so no disconnect_by_func until the bug is resolved.

Reply:
Ok, I will undo this commit